File: 1143-longest-common-subsequence/1143-longest-common-subsequence.py
from collections import Counter
class Solution(object):
    def longestCommonSubsequence(self, text1, text2):
        """
        :type text1: str
        :type text2: str
        :rtype: int
        """
        n = len(text1)
        m = len(text2)
        dp = [[0] * (m+1) for _ in range(n+1)]
        for i in range(1, n + 1):
            for j in range(1, m + 1):
                if text1[i - 1] == text2[j - 1]:
                    dp[i][j] = 1 + (dp[i - 1][j - 1])
                else:
                    dp[i][j] = max(dp[i][j - 1], dp[i - 1][j])
        return dp[n][m]

File: 1143-longest-common-subsequence/1143-longest-common-subsequence.java
class Solution {
  public int longestCommonSubsequence(String text1, String text2) {
    int M=text1.length();
    int N=text2.length();
    int[][] dp= new int[M+1][N+1];
    for (int i = 1; i <=M; i++) {
      for (int j = 1; j <=N; j++) {
        if(text1.charAt(i-1)==text2.charAt(j-1)){
          dp[i][j]=+1+dp[i-1][j-1];
        }else{
          dp[i][j]= Math.max(dp[i][j-1],dp[i-1][j]);
        }
      }
    }
    return dp[M][N];
  }
}